File: app/src/main/java/ca/anodsplace/headunit/utils/NetworkUtils.java
package ca.anodsplace.headunit.utils;

import android.content.Context;
import android.net.wifi.WifiManager;

import java.io.IOException;
import java.net.InetAddress;
import java.net.UnknownHostException;

/**
 * @author algavris
 * @date 06/11/2016.
 */

public class NetworkUtils {

    public static int getWifiIpAddress(Context context) throws IOException {
        final WifiManager wifiManager = (WifiManager) context.getApplicationContext().getSystemService(Context.WIFI_SERVICE);
        return wifiManager.getConnectionInfo().getIpAddress();
    }

    /**
     * Convert a IPv4 address from an integer to an InetAddress.
     * @param hostAddress an int corresponding to the IPv4 address in network byte order
     */
    public static InetAddress intToInetAddress(int hostAddress) {
        byte[] addressBytes = { (byte)(0xff & hostAddress),
                (byte)(0xff & (hostAddress >> 8)),
                (byte)(0xff & (hostAddress >> 16)),
                (byte)(0xff & (hostAddress >> 24)) };

        try {
            return InetAddress.getByAddress(addressBytes);
        } catch (UnknownHostException e) {
            AppLog.INSTANCE.e(e);
            throw new AssertionError();
        }
    }

    /**
     * Convert a IPv4 address from an InetAddress to an integer
     * @param inetAddr is an InetAddress corresponding to the IPv4 address
     * @return the IP address as an integer in network byte order
     */
    public static int inetAddressToInt(InetAddress inetAddr)
            throws IllegalArgumentException {
        byte [] addr = inetAddr.getAddress();
        return ((addr[3] & 0xff) << 24) | ((addr[2] & 0xff) << 16) |
                ((addr[1] & 0xff) << 8) | (addr[0] & 0xff);
    }

}

File: app/src/main/java/info/anodsplace/headunit/utils/NetworkUtils.kt
package info.anodsplace.headunit.utils

import android.content.Context
import android.net.wifi.WifiManager

import java.io.IOException
import java.net.InetAddress
import java.net.UnknownHostException

object NetworkUtils {
    fun getWifiIpAddress(context: Context): Int {
        val wifiManager = context.applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager
        return wifiManager.connectionInfo.ipAddress
    }

    /**
     * Convert a IPv4 address from an integer to an InetAddress.
     * @param hostAddress an int corresponding to the IPv4 address in network byte order
     */
    fun intToInetAddress(hostAddress: Int): InetAddress {
        val addressBytes = byteArrayOf((0xff and hostAddress).toByte(), (0xff and (hostAddress shr 8)).toByte(), (0xff and (hostAddress shr 16)).toByte(), (0xff and (hostAddress shr 24)).toByte())

        try {
            return InetAddress.getByAddress(addressBytes)
        } catch (e: UnknownHostException) {
            AppLog.e(e)
            throw AssertionError()
        }
    }
}
